### PR TITLE
Write manually-created conversion funcs for shared ComponentConfig types

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:

Due to how `conversion-gen` works today, it only discovers manually-generated conversion functions when creating autogenerated content for a referencing package. So in the example of `pkg/apis/componentconfig` referencing `k8s.io/apiserver/pkg/apis/config` types, no "native" conversion functions will be called from `pkg/apis/componentconfig/v1alpha1/zz_generated.conversion.go` unless these `Convert_*` functions exist in `k8s.io/apiserver/pkg/apis/config/v1alpha1/`.

We might consider this a bug in `conversion-gen`, I don't know, but to unblock the process of moving componentconfigs out of `pkg/apis/` I'm sending this PR so https://github.com/kubernetes/kubernetes/pull/66722 eventually can merge.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
ref: kubernetes/community#2354

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```
/assign @liggitt @sttts

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/apis/config/v1alpha1/conversion.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import (
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apiserver/pkg/apis/config"
+)
+
+// Important! The public back-and-forth conversion functions for the types in this generic
+// package with ComponentConfig types need to be manually exposed like this in order for
+// other packages that reference this package to be able to call these conversion functions
+// in an autogenerated manner.
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/apis/config/v1alpha1/conversion.go]: why is this? Is this a bug in conversion-gen? Or is it by design?

Comment (stewart-yu) [reply staging/src/k8s.io/apiserver/pkg/apis/config/v1alpha1/conversion.go]: The file creating Manually.
Like the comments, if no this file, `zz_conversion.go` in `pkg/apis` can't reference the `conversion function` in `k8s.io/<repo-name>`

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/apis/config/v1alpha1/conversion.go]: Am not convinced. This pretty much looks like a bug of conversion-gen.

Comment (luxas) [reply staging/src/k8s.io/apiserver/pkg/apis/config/v1alpha1/conversion.go]: It probably is, but I hope we can merge this to unblock the ComponentConfig move process

**Pull request merged**